### PR TITLE
Added example api code showing how to add auth tokens and user agents to prom client.

### PR DESCRIPTION
**Pull request by bwplotka**
Found some users looking for this, so adding to help others :hugs: 

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Review by beorn7 — CHANGES_REQUESTED** [review]
Awesome! But could you run gofmt on it?

File: api/prometheus/v1/example_test.go [hunk 1]
```diff
@@ -76,6 +78,118 @@ func ExampleAPI_queryRange() {
...
+		for k, s := range r.Header {
+			r2.Header[k] = s
+		}
+		r2.Header.Set("User-Agent", u.name)
+		r = r2
+	}
+	return u.rt.RoundTrip(r)
+}
+
+func ExampleAPI_queryRange_withUserAgent() {
```
Comment (beorn7) [inline api/prometheus/v1/example_test.go:104]: It appears double-underscoring doesn't work. `ExampleAPI_queryRangeWithUserAgent` should work.
Same for the below.

Comment (bwplotka) [reply api/prometheus/v1/example_test.go:104]: ack

**Comment by bwplotka** [comment]
done (:

**Review by beorn7 — APPROVED** [review]
Feel free to merge on green. (I'm offline for a while now.)

**Pull request merged**